File: src/main/java/ru/spb/sspk/ssdmd/phonebook/service/impl/PersonServiceImpl.java
package ru.spb.sspk.ssdmd.phonebook.service.impl;

import org.jvnet.hk2.annotations.Service;
import org.springframework.stereotype.Component;
import org.springframework.util.StringUtils;
import ru.spb.sspk.ssdmd.phonebook.model.dto.PersonDto;
import ru.spb.sspk.ssdmd.phonebook.model.entity.Person;
import ru.spb.sspk.ssdmd.phonebook.model.mapper.PersonMapper;
import ru.spb.sspk.ssdmd.phonebook.repository.PersonRepository;
import ru.spb.sspk.ssdmd.phonebook.service.PersonService;

import java.util.HashMap;
import java.util.List;
import java.util.Map;
import java.util.stream.Collectors;

@Service
@Component
public class PersonServiceImpl implements PersonService {

    private final PersonRepository personRepository;

    public PersonServiceImpl(PersonRepository personRepository) {
        this.personRepository = personRepository;
    }

    public List<Person> getAll() {
        return personRepository.findAll();
    }


    @Override
    public List<PersonDto> findPerson(String answer) {


        return null;
    }

    @Override
    public PersonDto save(String answer) {

        PersonDto person = null;

//        person = personMapper.toDto(personRepository.save(personMapper.toEntity(personDto)));

        return person;
    }

    @Override
    public PersonDto update(String answer) {

        PersonDto person = null;

//        person = personMapper.toDto(personRepository.update(personMapper.toEntity(personDto)));

        return person;
    }

    @Override
    public List<PersonDto> findAll() {

        List<PersonDto> personList = null;

        personList = personRepository.findAll().stream()
                .map(PersonMapper::toDto)
                .collect(Collectors.toList());

        return personList;
    }

    @Override
    public String findByAll(String answer) {

        List<PersonDto> personList = null;

        if (answer == null || answer.length() == 0) {
            return "не удалось выполнить запрос";
        } else {
            String firstName = StringUtils.capitalize(answer.toLowerCase());
            String lastName = StringUtils.capitalize(answer.toLowerCase());
            String department = answer.toUpperCase();

            Map<String, Object> paramMap = new HashMap<>(16);
            paramMap.put("first_name", firstName);
            paramMap.put("last_name", lastName);
            paramMap.put("department", department);

            personList = personRepository.findByAll(paramMap).stream()
                    .map(PersonMapper::toDto)
                    .collect(Collectors.toList());
        }

            return personList.toString().replace("[", "")
                    .replace("]", "")
                    .replace(",", ",\n\n");
    }

    @Override
    public String deleteById(Long id) {

        personRepository.deleteById(id);

        return "managed to complete";
    }
}

File: src/main/java/ru/spb/sspk/ssdmd/phonebook_test/service/PersonServiceImpl.kt
package ru.spb.sspk.ssdmd.phonebook_test.service

import org.springframework.stereotype.Component
import ru.spb.sspk.ssdmd.phonebook_test.model.dto.PersonDto
import ru.spb.sspk.ssdmd.phonebook_test.model.mapper.PersonMapper
import ru.spb.sspk.ssdmd.phonebook_test.repository.PersonRepository
import java.util.*

@Component
class PersonServiceImpl(
    private val personRepository: PersonRepository,
    private val personMapper: PersonMapper
) : PersonService {

    override fun performingSearchByAllParameters(answer: String): String {
        val lastname: String = if ('-' in answer) {
            answer.lowercase()
            answer.split('-').joinToString("-") {
                it.replaceFirstChar {if (it.isLowerCase()) it.titlecase() else it.toString()}}
        } else answer.lowercase()
            .replaceFirstChar { if (it.isLowerCase()) it.titlecase(Locale.getDefault()) else it.toString() }

        val firstname = answer.lowercase()
            .replaceFirstChar { if (it.isLowerCase()) it.titlecase(Locale.getDefault()) else it.toString() }
        val department = answer.uppercase(Locale.getDefault())
        val block = answer.uppercase(Locale.getDefault())
        val sector = answer.uppercase(Locale.getDefault())
        val phone = if (answer.all { it.isDigit() }) answer
        else null

        val mobilPhone = if (answer.any() { it.isDigit() }) {
            if (answer.startsWith("8")) answer.removePrefix("8")
            else if (answer.startsWith("+7")) answer.removePrefix("+7")
            else answer.removePrefix("7")
        } else null
        val persons: List<PersonDto> =
            personRepository.findByFirstnameOrLastnameOrDepartmentOrBlockOrSectorOrMobilPhoneOrPhone(
                firstname,
                lastname,
                department,
                block,
                sector,
                mobilPhone,
                phone
            ).map { person -> personMapper.toDto(person) }.toList()
        return persons.toString().removeSurrounding("[", "]").replace(",", ",\n\n")
    }
}